jujugui/static/gui/src/app/components/service-overview/test-service-overview.js

```diff
@@ -71,14 +71,50 @@ describe('ServiceOverview', function() {
     var none = undefined;
     assert.deepEqual(output.props.children[0],
       <juju.components.OverviewAction
+        icon={none}
         key="Units"
         title="Units"
         value={value}
-        icon={none}
-        action={none}
         valueType={none}
         link={none}
-        linkTitle={none} />);
+        linkTitle={none}
+        action={output.props.children[0].props.action} />);
+  });
+
+  it('clicks to All Units line item navigates to the unit list', function() {
+    var getStub = sinon.stub();
+    getStub.withArgs('units').returns({toArray: function() {
+      return [{}, {}];
+    }});
+    getStub.withArgs('id').returns('demo');
+    var service = {
+      get: getStub
+    };
+    var changeState = sinon.stub();
+    var shallowRenderer = testUtils.createRenderer();
+    stubIcons();
```

Comment:
I guess having stuff like this means we can't always use the shallowRender util.